feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts

```diff
@@ -130,14 +132,16 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
     this.subscription.add(
       this.launchDialogService?.dialogClose
         .pipe(filter((result) => Boolean(result)))
-        .subscribe((result: CustomerListAction) => {
+        .subscribe((result) => {
```

Comment:
At first, I thought the same, but the compiler will prompt and report errors, which as shown in the screenshot.
![image](https://user-images.githubusercontent.com/36021549/220849142-24dec3f3-902b-4061-9468-eefa8bda947a.png)

Reply:
This is because the expression in the `if()` statement does not effectively narrow the type of `result`.  See [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing)

Reply:
Cool, you are right.
By the way, I also found a other way that just add a type in the if() statement . Which one way is better? As shown in the screenshot below
![image](https://user-images.githubusercontent.com/126219214/221135494-58a8933f-bc0a-40cd-936c-9cc110a1f900.png)
![image](https://user-images.githubusercontent.com/126219214/221136907-294b60fa-70af-457b-944e-6308572dbb87.png)

Reply:
The first option (the one without `any`) is best.  Generally, `any` should be avoided, as it causes most type checking to be ignored, which defeats the point of using typescript.

For this case, adding `any` to the union will erases all the types.  If you look at the inferred type of `result` in the blocks (by hovering over the symbol), you will see that it is set to `any` instead of `CustomerListAction` or `CreatedCustomer`.

Reply:
changed